pkg/plugin/dv_backup_item_action.go

```diff
@@ -74,51 +78,90 @@ func (p *DVBackupItemAction) Execute(item runtime.Unstructured, backup *v1.Backu
 
 	extra := []velero.ResourceIdentifier{}
 
+	kind := item.GetObjectKind().GroupVersionKind().Kind
+	switch kind {
+	case "PersistentVolumeClaim":
+		return p.handlePVC(item)
+	case "DataVolume":
+		return p.handleDataVolume(backup, item)
+	}
+
+	return item, extra, nil
+}
+
+func (p *DVBackupItemAction) handlePVC(item runtime.Unstructured) (runtime.Unstructured, []velero.ResourceIdentifier, error) {
 	metadata, err := meta.Accessor(item)
 	if err != nil {
 		return nil, nil, err
 	}
+	p.log.Infof("handling PVC %v/%v", metadata.GetNamespace(), metadata.GetName())
 
-	annotations := metadata.GetAnnotations()
-	if annotations == nil {
-		annotations = make(map[string]string)
+	dv, err := p.getOwningDataVolume(metadata)
+	if err != nil {
+		return nil, nil, err
 	}
-
-	kind := item.GetObjectKind().GroupVersionKind().Kind
-	switch kind {
-	case "PersistentVolumeClaim":
-		annotations = p.handlePVC(annotations, metadata)
-	case "DataVolume":
-		annotations, extra = p.handleDataVolume(annotations, metadata)
+	if dv != nil {
+		annotations := metadata.GetAnnotations()
+		if annotations == nil {
+			annotations = make(map[string]string)
+		}
+		if dv.Status.Phase == cdiv1.Succeeded {
+			// make sure an object is marked as populated, so the operation will not be retried after restore
+			annotations[AnnPopulatedFor] = dv.Name
+		} else {
+			// The PVC is not finished, we mark it as inprogress, so it can be skipped during restore
+			// so it does not conflict with CDI action
+			annotations[AnnInProgress] = dv.Name
+		}
+		metadata.SetAnnotations(annotations)
 	}
 
-	metadata.SetAnnotations(annotations)
-
+	extra := []velero.ResourceIdentifier{}
 	return item, extra, nil
 }
 
-func (p *DVBackupItemAction) handleDataVolume(annotations map[string]string, metadata metav1.Object) (map[string]string, []velero.ResourceIdentifier) {
-	p.log.Infof("handling DataVolume %v/%v", metadata.GetNamespace(), metadata.GetName())
-	// TODO: if DV name will ever be different that PVC name, this must be changed
-	annotations[AnnPrePopulated] = metadata.GetName()
+func (p *DVBackupItemAction) handleDataVolume(backup *v1.Backup, item runtime.Unstructured) (runtime.Unstructured, []velero.ResourceIdentifier, error) {
+	var dv cdiv1.DataVolume
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(item.UnstructuredContent(), &dv); err != nil {
+		return nil, nil, errors.WithStack(err)
+	}
 
-	extra := []velero.ResourceIdentifier{{
-		GroupResource: kuberesource.PersistentVolumeClaims,
-		Namespace:     metadata.GetNamespace(),
-		Name:          metadata.GetName(),
-	}}
+	p.log.Infof("handling DataVolume %v/%v", dv.GetNamespace(), dv.GetName())
+	extra := []velero.ResourceIdentifier{}
+	dvSucceeded := dv.Status.Phase == cdiv1.Succeeded
+	if dvSucceeded {
+		annotations := dv.GetAnnotations()
+		if annotations == nil {
+			annotations = make(map[string]string)
+		}
+		annotations[AnnPrePopulated] = dv.GetName()
```

Comment:
Ultimately doesn't matter now but we should probably use `status.claimName` if set

Reply:
Thanks for info, 
ok, I need to update the CDI used in imports, and probably switch to CDI-API. But that is work for another PR. 
https://github.com/kubevirt/kubevirt-velero-plugin/issues/69